app/code/Magento/ConfigurableProduct/Model/ConfigurableAttributeData.php

```diff
@@ -48,6 +66,35 @@ public function getAttributesData(Product $product, array $options = [])
         ];
     }
 
+    /**
+     * Get configurable product attribute position
+     *
+     * @param $attribute
+     *
+     * @return mixed
+     */
+    protected function getPosition($attribute)
+    {
+        if ($this->getConfigurableProductAttributesSortConfig() === ProductAttributesSort::GLOBAL_ATTRIBUTE_SETTING) {
+            return $attribute->getProductAttribute()->getPosition();
+        }
+
+        return $attribute->getPosition();
+    }
+
+    /**
+     * Get configuration value for configurable products attributes sorting settings
+     *
+     * @return int
+     */
+    protected function getConfigurableProductAttributesSortConfig()
```

Comment:
Please change the protected property to private. According [Magento technical guideline](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html)  we can't introduce new protected properties